src/Microsoft.Health.Fhir.Core/Features/Operations/Reindex/ReindexJobTask.cs

```diff
@@ -283,15 +283,34 @@ public async Task ExecuteAsync(ReindexJobRecord reindexJobRecord, WeakETag weakE
                         }
                     }
 
-                    // if our received CancellationToken is cancelled we should
+                    // Get the latest version of the reindex job in case another thread has updated it
+                    await jobSemaphore.WaitAsync();
+                    try
+                    {
+                        using (IScoped<IFhirOperationDataStore> store = _fhirOperationDataStoreFactory())
+                        {
+                            var wrapper = await store.Value.GetReindexJobByIdAsync(_reindexJobRecord.Id, cancellationToken);
+                            _weakETag = wrapper.ETag;
```

Comment:
>_weakETag = wrapper.ETag; [](http://example.com/codeflow?start=28&length=25)

Why this is necessary? In case if no one put an eye on this job, etag would be the same, and in case job been cancelled tag doesn't matter anymore.

Reply:
It is true.  This is more of a just in case type check and update.